### ISSUE TITLE
Non-unique ids bug

### ISSUE THREAD

**Issue by jackens** [tgreyuk/typedoc-plugin-markdown#303]
`typedoc.json`:

```json
{
  "cleanOutputDir": true,
  "entryPointStrategy": "expand",
  "entryPoints": [
    "src"
  ],
  "hideBreadcrumbs": true,
  "hideInPageTOC": true,
  "hideLegend": true,
  "namedAnchors": true,
  "out": "_doc",
  "readme": "none"
}
```

`tsconfig.json`:

```json
{
  "compilerOptions": {
    "allowJs": true,
    "module": "commonjs",
    "outDir": "tmp",
    "target": "esnext"
  },
  "include": [
    "src/**/*.js"
  ]
}
```

`package.json`:

```json
{
  "dependencies": {
    "typedoc": "0.22.13",
    "typedoc-plugin-markdown": "3.11.14",
    "typescript": "4.6.3"
  },
  "name": "non-unique-ids-bug",
  "type": "module"
}
```

`src/file.js`

```js
/**
 * @typedef {() => number} ANSWER
 */

/**
 * @type ANSWER
 */
export const answer = () => 42
```

Executing

```sh
npm i
npx typedoc
```

generates the following `_doc/README.md` file with multiple `id="answer"`:

~~~md
# non-unique-ids-bug

## Type aliases

### <a id="answer" name="answer"></a> ANSWER

Ƭ **ANSWER**<\>: () => `number`

#### Type declaration

▸ (): `number`

##### Returns

`number`

#### Defined in

file.js:2

## Functions

### <a id="answer" name="answer"></a> answer

▸ **answer**(): `number`

#### Returns

`number`

#### Defined in

file.js:8
~~~

**Comment by tgreyuk** [comment]
Thanks for detailed description. Fix in typedoc-plugin-markdown@3.12.0.

**Comment by KillyMXI** [comment]
I've got a false positive with 3.12.0 - links to a unique heading got an index added.
I think that will not work on GitHub so I'm staying on `~3.11.14` for now.

Inside the file https://github.com/mxxii/parseley/blob/main/docs/modules/Ast.md it creates the following link
``[`Specificity`](Ast.md#specificity-6)``
for the unique heading.

**Comment by tgreyuk** [comment]
@KillyMXI thanks - i see the issue - it should now be fixed (typedoc-plugin-markdown@3.12.1)

**Comment by KillyMXI** [comment]
Thank you. 3.12.1 works fine.
